src/hearings/containers/request-hearing/date-priority-hearing/date-priority-hearing.component.ts

```diff
@@ -0,0 +1,80 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+import { GovUiConfigModel } from '@hmcts/rpx-xui-common-lib/lib/gov-ui/models';
+import { RefDataModel } from 'api/hearings/models/refData.model';
+
+
+@Component({
+  selector: 'exui-date-priority-hearing',
+  templateUrl: './date-priority-hearing.component.html',
+  styleUrls: ['./date-priority-hearing.component.scss']
+})
+export class DatePriorityHearingComponent implements OnInit {
+  public priorityForm: FormGroup;
+  public priorities: RefDataModel[];
+  public checkedHearingAvailability: string;
+  public partiesNotAvailableDates: string[];
+  public firstHearingDate: GovUiConfigModel;
+  public earliestHearingDate: GovUiConfigModel;
+  public latestHearingDate: GovUiConfigModel;
+
+  constructor(private readonly formBuilder: FormBuilder,
+              private readonly route: ActivatedRoute) { }
+
+  public ngOnInit(): void {
+    this.initDateConfig();
+    this.initForm();
+
+    this.priorities = this.route.snapshot.data.hearingPriorities.priorities.sort((a, b) => (a.order < b.order ? -1 : 1));
+
+    // TODO: Get dates from a service
+    this.partiesNotAvailableDates = ['2 November 2021', '3 November 2021', '4 November 2021'];
+  }
+
+  /**
+   * Inits date config
+   */
```

Comment:
unnecessary comment the method name is already explanatory and the comment is explaining no more than the name off the method. i.e. code smell